### PR TITLE
drivers: pinctrl: rename NXP Kinetis pinctrl to NXP PORT, and consolidate headers

### PR DESCRIPTION
**Pull request by danieldegrasse**
The NXP Kinetis series uses the PORT peripheral for pin control. At the time the driver was introduced, this IP was only used on the Kinetis series in tree. However, it is now used on NXP S32, MCX, and Kinetis series SOCs. Rename the pin control driver compatible and files to use the naming scheme `nxp_port`, which better reflects the name of the IP in the reference manual.

This PR also consolidates the header files used for pin control with this driver, by checking if `PCR_XXX` bitfields are defined as a method of determining which pin configuration features a given PORT IP instantiation supports.

### CODE REVIEW

**Comment by danieldegrasse** [comment]
@hakehuang I believe we should run some basic hardware tests on this PR, covering all Kinetis and MCX targets (as well as S32, if possible). We likely just need to verify hello world builds and runs to make sure that the pin control driver is functioning as expected on these parts.

**Review by decsny — DISMISSED** [review]

**Review by manuargue — DISMISSED** [review]
💯

**Comment by hakehuang** [comment]
> believe we should run some basic hardware tests on this PR, covering all Kinetis and MCX targets (as well as S32, if possible). We likely just need to verify hello world builds and runs to make sure that the pin control driver is functioning as expected on these parts.

@danieldegrasse  I do not have s32 paltform sin my test range, all BLEP paltform test are kicked.

File: doc/releases/migration-guide-4.1.rst [hunk 1]
```diff
@@ -60,6 +60,12 @@ Interrupt Controller
 LED Strip
 =========
 
+Pin Control
+===========
+
+  * Renamed the ``compatible`` from ``nxp,kinetis-pinctrl`` to :dtcompatible:`nxp,port-pinctrl`.
+  * Renamed the ``compatible`` from ``nxp,kinetis-pinmux`` to :dtcompatible:`nxp,port-pinmux`.
```
Comment (butok) [inline doc/releases/migration-guide-4.1.rst:67]: We can have it shorter, without "port":
```
- nxp,pinctrl
- nxp,pinmux
- CONFIG_PINCTRL_NXP
```

**Review by gmarull — COMMENTED** [review]
do all modified socs share the same IP? If not, compatibles must be different, and instead, make pinctrl driver support multiple of them.

**Review by gmarull — DISMISSED** [review]

File: west.yml [hunk 2]
```diff
@@ -198,7 +198,7 @@ manifest:
       groups:
         - hal
     - name: hal_nxp
-      revision: 3c64cd63125c86870802a561ce79dc33697b005c
+      revision: pull/471/head
```
Comment (mmahadevan108) [inline west.yml:201]: Please update the SHA

**Comment by danieldegrasse** [comment]
PR has been rebased, and updated to pull SHA from NXP HAL

**Review by decsny — DISMISSED** [review]

**Review by decsny — APPROVED** [review]

**Review by dleach02 — APPROVED** [review]

**Comment by dleach02** [comment]
@gmarull I added myself to the assignee so we can help move this PR to merge. You had already approved it but we lost all the approvals when we updated the manifest with the HAL PR merging. We have some HAL side PRs we are trying to sequence.

**Review by mmahadevan108 — APPROVED** [review]

**Pull request merged**